internal/drivers/amazon/driver.go

```diff
@@ -360,7 +361,7 @@ func (p *config) Create(ctx context.Context, opts *types.InstanceCreateOpts) (in
 }
 
 // Destroy destroys the server AWS EC2 instances.
-func (p *config) Destroy(ctx context.Context, instances []*types.Instance) (err error) {
+func (p *config) Destroy(ctx context.Context, instances []*types.Instance, storageCleanupType *storage.CleanupType) (err error) {
```

Comment:
Instead of setting nil wherever this is not required, we can overload the method, and set nil in the overloaded method, using a common method internally.

Reply:
Updated